src/ui/inspector/PlotElementsPool.vue

```diff
@@ -0,0 +1,316 @@
+                    return true;
+                }
+            }
+
+            return false;
+        }
+    },
+    mounted() {
+        const selection = this.openmct.selection.get();
+        if (selection && selection.length > 0) {
+            this.showSelection(selection);
+        }
+
+        this.openmct.selection.on('change', this.showSelection);
+        this.openmct.editor.on('isEditing', this.setEditState);
+    },
+    destroyed() {
+        this.openmct.editor.off('isEditing', this.setEditState);
+        this.openmct.selection.off('change', this.showSelection);
+
+        this.unlistenComposition();
+    },
+    methods: {
+        setEditState(isEditing) {
+            this.isEditing = isEditing;
+            this.showSelection(this.openmct.selection.get());
+        },
+        showSelection(selection) {
+            if (_.isEqual(this.selection, selection)) {
+                return;
+            }
+
+            this.selection = selection;
+            this.elementsCache = {};
+            this.listeners = [];
+            this.parentObject = selection && selection[0] && selection[0][0].context.item;
+            this.setYAxisIds();
+
+            this.unlistenComposition();
+
+            if (this.parentObject) {
+                this.composition = this.openmct.composition.get(this.parentObject);
+
+                if (this.composition) {
+                    this.composition.load();
+                    this.registerCompositionListeners();
+                }
+            }
+        },
+        unlistenComposition() {
+            if (this.compositionUnlistener) {
+                this.compositionUnlistener();
+            }
+        },
+        registerCompositionListeners() {
+            this.composition.on('add', this.addElement);
+            this.composition.on('remove', this.removeElement);
+            this.composition.on('reorder', this.reorderElements);
+
+            this.compositionUnlistener = () => {
+                this.composition.off('add', this.addElement);
+                this.composition.off('remove', this.removeElement);
+                this.composition.off('reorder', this.reorderElements);
+                delete this.compositionUnlistener;
+            };
+        },
+        setYAxisIds() {
+            const configId = this.openmct.objects.makeKeyString(this.parentObject.identifier);
+            this.config = configStore.get(configId);
+            this.yAxes.push({
+                id: this.config.yAxis.id,
+                elements: this.parentObject.configuration.series.filter(
+                    series => series.yAxisId === this.config.yAxis.id
+                )
+            });
+            if (this.config.additionalYAxes) {
+                this.config.additionalYAxes.forEach(yAxis => {
+                    this.yAxes.push({
+                        id: yAxis.id,
+                        elements: this.parentObject.configuration.series.filter(
```

Comment:
Can elements be an empty Array here since `applySearch` assigns elements anyway?